UI/MainMenu/LoadGameScreen.gd

```diff
@@ -19,39 +22,34 @@ func _ready():
 	if OS.get_name() == "HTML5" and OS.has_feature("JavaScript"):
 		_define_js()
 
-func updateSaves():
-	Util.delete_children(savesContainer)
-	
+func updateSaves(): #Argh!!! It's was so unefficient...
 	var savesPaths = SAVE.getSavesSortedByDate()
 	
-	for savePath in savesPaths:
+	for save in savesContainer.get_children():
+		var path = save.getSavePath()
+		if path in savesPaths:
+			savesPaths.erase(path) # Removing existiong ones from check list - they aren't needed...
+		else:
+			save.queue_free() # ...And deleting saves that not existed!
+			
+	for savePath in savesPaths: # Then we load ONLY necessary ones!
 		var saveGameElementObject = saveGameElemenetScene.instance()
 		savesContainer.add_child(saveGameElementObject)
 		saveGameElementObject.setSaveFile(savePath)
 		saveGameElementObject.connect("onLoadButtonPressed", self, "onSaveLoadButtonClicked")
 		saveGameElementObject.connect("onDeleteButtonPressed", self, "onDeleteButtonClicked")
 		saveGameElementObject.connect("onExportButtonPressed", self, "onExportButtonClicked")
+		# Saves know when change to delete mode! We now can call this func less often!
+		var _ok = connect("changeDeleteMode", saveGameElementObject, "setDeleteMode")
 		saveGameElementObject.setDeleteMode(inDeleteMode)
 		
+		UIbuttonsCon.visible = !(is_instance_valid(GM.ui) && GM.ui.main_menu_open)
```

Comment:
Huh? This looks very strange

Reply:
In Main Menu I use SceneButtons to interact with menu. On In-Game menu, in opposite, it should be shown.